tests/test_utilities.h

```diff
@@ -8,14 +8,14 @@
 
 namespace testing {
 
-std::ofstream deallogfile;
+std::ofstream deallogstream;
```

Comment:
Should we have a class managing this resource? 

Also, declaring it at namespace scope could cause hard to find runtime errors if we reference it in other translation units (because the relative order of initialization of non-local static variables in different translational units is not defined).

Reply:
That sounds overcomplicating things to me. Normally the ofstream object is only be invoked `init_log` and the MPI variant of it. But it is also useful when it is preferred over deallog in some cases even when deallog is in use, e.g. in some lines of a log you want the prefix but in some other lines you don't want.

Reply:
But we could still talk about the redesign of the stream related stuff and do another PR if we have an agreement on the new design.

Reply:
We might want to in the future. My biggest concern is that calling: `testing::deallogstream` in any other `.cc` file has a 50%/50% chance of causing a runtime error that won't be caught by the compiler. [`static` initialization order fiasco](https://isocpp.org/wiki/faq/ctors#static-init-order)

I don't anticipate us doing that, but leaving ourselves open to runtime errors that we can avoid is probably a best practice. Let's revisit in the future!